model/roles.go

```diff
@@ -380,9 +381,61 @@ func (m *RoleManifest) GetRoleManifestDevPackageVersion(roles Roles, opinions *O
 
 // LookupRole will find the given role in the role manifest
 func (m *RoleManifest) LookupRole(roleName string) *Role {
+	if m.rolesByName == nil {
```

Comment:
I would drop the `rolesByName` cache and do a search for every lookup. I don't think you'll ever even have a 100 roles in a single manifest, so this feels like premature optimization.

If you want to keep the cache, then maybe extend the condition to be robust against manifest changes after the first lookup:

```
if m.RolesByName == nil || m.RolesByName[role.Name] == nil {
```

That of course assumes that most lookups will be successful. Because you'll rebuild the cache on each failed lookup. But I would just drop the cache. 😃